shared/src/zeroex_api.rs

```diff
@@ -68,12 +64,13 @@ impl SwapQuery {
             url.query_pairs_mut()
                 .append_pair("buyAmount", &amount.to_string());
         }
-        if let Some(skip_validation) = self.skip_validation {
-            url.query_pairs_mut()
-                .append_pair("skipValidation", &skip_validation.to_string());
-        }
         url.query_pairs_mut()
             .append_pair("affiliateAddress", AFFILIATE_ADDRESS);
+        // We do not provide a takerAddress so validation does not make sense.
+        url.query_pairs_mut().append_pair("skipValidation", "true");
+        // Ensure that we do not request binding quotes that we might be penalized for not taking.
+        url.query_pairs_mut()
+            .append_pair("intentOnFilling", "false");
```

Comment:
This is true for price estimates, but not for the single order solver right?

Reply:
It should not because for the single order solver we do not guarantee that the quote will actually get executed. After https://github.com/gnosis/gp-v2-services/issues/828 what we could do is use price endpoint in the single order solver with RFQT liquidity enabled and only use quote if its settlement wins.

Reply:
Technically `intentOnFilling` does nothing for the price endpoint anyway but I felt it was safer to unconditionally include it for now.

Reply:
I see, its the same issue we have with ParaSwap I guess. I guess we don't have access to their RFQT liquidity then?

Reply:
We shouldn't have access to their RFQT liquidity in the price call because otherwise we might quote a better price and gas amount than what we will get later through quote (which is definitely not using RFQT since intentOnFilling=false). Their api docs aren't completely clear whether /prices includes this liquidity or not which is another reason for always having intentOnFilling=false.

Reply:
Relevant docs https://0x.org/docs/api#get-swapv1price and https://0x.org/docs/guides/rfqt-in-the-0x-api .

Actually reading through this again I think we shouldn't even use the price endpoint at the moment because they say it is meant for RFQT.

Reply:
I'm going to make another PR to tackle this in the price estimator instead of including it in this PR.

Reply:
Also, it looks like we never qualified for the RFQT liquidity:
> In order to qualify for RFQ-T liquidity, the request must include the query parameter intentOnFilling=true (in addition to the aforementioned 0x-api-key and non-null takerAddress).

So I don't have any concerns with keeping `intentOnFilling=false` until we implement #828